lending/loan_management/doctype/loan_adjustment/loan_adjustment.py

```diff
@@ -44,6 +46,39 @@ def validate(self):
 					},
 				)
 
+		available_sd = float(amounts.get("available_security_deposit") or 0)
+
+		precision = cint(frappe.db.get_default("currency_precision")) or 2
+
+		total_net_payable = round(
+			float(amounts.get("unaccrued_interest") or 0)
+			+ float(amounts.get("interest_amount") or 0)
+			+ float(amounts.get("penalty_amount") or 0)
+			+ float(amounts.get("total_charges_payable") or 0)
+			- float(amounts.get("available_security_deposit") or 0)
+			+ float(amounts.get("unbooked_interest") or 0)
+			+ float(amounts.get("unbooked_penalty") or 0)
+			+ float(amounts.get("pending_principal_amount") or 0),
+			precision,
+		)
+
+		total_adjustment_amount = sum(float(row.amount) for row in self.adjustments if row.amount)
+
+		sd_adjustment = next(
+			(row for row in self.adjustments if row.loan_repayment_type == "Security Deposit Adjustment"),
+			None,
+		)
```

Comment:
You are looping twice. Can be done in one single loop
Also use better variable names not `sd_adjustment` (better would be loan_security_deposit or available_security_deposit)